src/main/java/org/springframework/data/couchbase/repository/query/StringN1qlQueryCreator.java

```diff
@@ -81,8 +84,15 @@ public StringN1qlQueryCreator(final ParameterAccessor accessor, CouchbaseQueryMe
 		} else {
 			throw new IllegalArgumentException("query has no inline Query or named Query not found");
 		}
+		Class javaType = getType();
+		String typeValue = javaType.getName();
+		TypeInformation<?> typeInfo = ClassTypeInformation.from(javaType);
+		Alias alias = couchbaseConverter.getTypeAlias(typeInfo);
+		if (alias != null && alias.isPresent()) {
+			typeValue = alias.toString();
+		}
 		this.queryParser = new StringBasedN1qlQueryParser(queryString, queryMethod, bucketName, couchbaseConverter,
-				getTypeField(), getTypeValue(), accessor, spelExpressionParser, evaluationContextProvider);
+				getTypeField(), typeValue, accessor, spelExpressionParser, evaluationContextProvider);
```

Comment:
This had just been using the classname. It needs to use TypeAlias annotation if present.